### PR TITLE
Update Dart grammar for extension type highlighting and other fixes

### PR DESCRIPTION
**Pull request by parlough**
Updates from `v1.2.0` to `v1.2.5` from https://github.com/dart-lang/dart-syntax-highlight/blob/master/grammars/dart.json, with the primary change being support for highlighting `extension type` as a combined keyword and various comment fixes.

Complete changelog:

> ## 1.2.5 (2023-11-27)
> 
> - Don't mark empty double backticks as inline code.
> 
> ## 1.2.4 (2023-11-26)
> 
> - Include the opening `$` of a simple identifier string interpolation
>   inside the `string.interpolated.expression.dart` scope.
> 
> ## 1.2.3 (2023-09-11)
> 
> - Removed support for `inline class` keyword and added support for `extension type`.
> 
> ## 1.2.2 (2023-06-28)
> 
> - Added support for `inline class`.
> 
> ## 1.2.1 (2023-05-04)
> 
> - Fixed highlighting of comments inside type arguments.

### CODE REVIEW

**Review by bkonyi — APPROVED** [review]
RSLGTM

**Comment by bkonyi** [comment]
FYI @kenzieschmoll

File: packages/devtools_app/release_notes/NEXT_RELEASE_NOTES.md [hunk 1]
```diff
@@ -37,7 +37,9 @@ TODO: Remove this section if there are not any general updates.
 
 ## Debugger updates
 
-TODO: Remove this section if there are not any general updates.
+* Highlight `extension type` as a declaration keyword,
+  highlight the `$` in identifier interpolation as part of the interpolation,
+  and properly highlight comments within type arguments.
```
Comment (kenzieschmoll) [inline packages/devtools_app/release_notes/NEXT_RELEASE_NOTES.md:42]: please link to this PR in this release notes entry

**Review by kenzieschmoll — APPROVED** [review]
one comment and lgtm

File: packages/devtools_app/release_notes/NEXT_RELEASE_NOTES.md (hunk 1 above)
Comment (kenzieschmoll) [inline packages/devtools_app/release_notes/NEXT_RELEASE_NOTES.md:42]:
```suggestion
  and properly highlight comments within type arguments. - [6837](https://github.com/flutter/devtools/pull/6837)
```

**Pull request merged**